RecoLocalCalo/HGCalRecProducers/python/hgcalLayerClusters_cff.py

```diff
@@ -6,29 +6,37 @@
 
 from RecoLocalCalo.HGCalRecProducers.HGCalUncalibRecHit_cfi import HGCalUncalibRecHit
 
-from SimCalorimetry.HGCalSimProducers.hgcalDigitizer_cfi import fC_per_ele, hgceeDigitizer, hgchebackDigitizer, hfnoseDigitizer
+from SimCalorimetry.HGCalSimProducers.hgcalDigitizer_cfi import fC_per_ele, HGCAL_noises, hgceeDigitizer, hgchebackDigitizer, hfnoseDigitizer
 
-hgcalLayerClusters = hgcalLayerClusters_.clone()
-
-hgcalLayerClusters.timeOffset = hgceeDigitizer.tofDelay
-hgcalLayerClusters.plugin.dEdXweights = cms.vdouble(dEdX.weights)
-hgcalLayerClusters.plugin.fcPerMip = cms.vdouble(HGCalUncalibRecHit.HGCEEConfig.fCPerMIP)
-hgcalLayerClusters.plugin.thicknessCorrection = cms.vdouble(HGCalRecHit.thicknessCorrection)
-hgcalLayerClusters.plugin.fcPerEle = cms.double(fC_per_ele)
-hgcalLayerClusters.plugin.noises = cms.PSet(refToPSet_ = cms.string('HGCAL_noises'))
-hgcalLayerClusters.plugin.noiseMip = hgchebackDigitizer.digiCfg.noise
+hgcalLayerClusters = hgcalLayerClusters_.clone(
+    timeOffset = hgceeDigitizer.tofDelay,
+    plugin = dict(
+        dEdXweights = dEdX.weights.value(),
+        #With the introduction of 7 regional factors (6 for silicon plus 1 for scintillator),
+        #we extend fcPerMip (along with noises below) so that it is guaranteed that they have 6 entries.
+        fcPerMip = HGCalUncalibRecHit.HGCEEConfig.fCPerMIP.value() + HGCalUncalibRecHit.HGCHEFConfig.fCPerMIP.value(),
+        thicknessCorrection = HGCalRecHit.thicknessCorrection.value(),
+        sciThicknessCorrection = HGCalRecHit.sciThicknessCorrection.value(),
+        deltasi_index_regemfac = HGCalRecHit.deltasi_index_regemfac.value(),
+        fcPerEle = fC_per_ele,
+        #Extending noises as fcPerMip, see comment above.
+        noises = HGCAL_noises.values.value() + HGCAL_noises.values.value(),
+        noiseMip = hgchebackDigitizer.digiCfg.noise.value()
+    )
+)
 
```

Comment:
One of these two empty lines could be removed, both for the aesthetics and also to get rid of spurious differences with what in the master